File: first_wave/pyth/ex2.py


import argparse
import os


class Box:
    xSize=0
    ySize=0
    rule="none"
    def __init__(self, x, y, rule):
        self.xSize=x; self.ySize=y
        self.rule = rule
    
    def spaceChecker(self, x, y):
        if( (x+y) % 2 == 0):
            print("*", end="")
            return
        print(" ", end="")
        return

    def updateBox(self):
        for i in range(int(self.ySize)):
            if i==0 or i==int(self.ySize)-1:
                for k in range(int(self.xSize)):
                    print("_", end="")
                print("")
                continue
            for j in range(int(self.xSize)):
                if j==0 or j==int(self.xSize)-1:
                    print("|", end="")
                    if j==int(self.xSize)-1:
                        print("")
                    continue
                else:
                    self.spaceChecker(i, j)

     




def main():
    xSize=0
    ySize=0
    print("Welcome to the box printer! Please introduce x and y dimensions: ")
    xSize=input()
    ySize=input()
    myBox = Box(xSize, ySize, rule="even2*");
    os.system("clear")
    myBox.updateBox()


if __name__=='__main__':
    main()

File: first_wave/lua/ex2.lua

local function WhichDraw(entity, xPosition, yPosition)
    if (xPosition + yPosition) % 2 == 0 then
        io.write("*")
        return;
    end
    io.write(entity.boxSpace)

end


local function Draw(entity)
    --io.write(tonumber(entity.ySize))
    for j = 0, tonumber(entity.ySize), 1 do
        if (j==0 or j==entity.ySize) then
            for i=0, tonumber(entity.xSize), 1 do
                io.write("-")
            end
            ---io.write("\n")
            if j == entity.ySize or j==0 then
                io.write("\n")
            end
            goto continue
        end
        
        for i = 0, tonumber(entity.xSize), 1 do 
            if (i==0 or i==entity.xSize) then
                io.write(entity.verticalWall)
                
            end
            if not (i==0 or i==entity.ySize) then
                WhichDraw(entity, i, j)    
            end
            if i == entity.xSize then
                io.write("\n")
            end
            
        end
        ::continue::
        
    end
end



local function Box(xSize, ySize)
    local xSize = xSize or 10
    local ySize = ySize or 10
    

    
    return{
        xPosition=0,
        yPosition=0,
        boxSpace=' ',
        xSize = xSize,
        ySize=ySize,
        verticalWall = '|',
        horizontalWall = '-'
    }
end


io.write("Welcome to the box printer! Please introduce X and Y dimension:\n")
local xSize = tonumber(io.read())
local ySize = tonumber(io.read())
local start = os.clock()
_G.myBox = Box(xSize, ySize)
Draw(myBox)
print("\n", os.clock() - start)
